client/templates/layouts/default.js

```diff
@@ -1,30 +1,27 @@
-const handleRedirect = ( routes, redirect ) => {
-	let currentRoute = FlowRouter.getRouteName();
-	if ( routes.indexOf( currentRoute ) > -1 ) {
-		FlowRouter.go( redirect );
-		return true;
-	}
+const authenticated = () => {
+  return !Meteor.loggingIn() && Meteor.user();
+}
+
+const handleRedirect = (currentRoute) => {
+  let authData = currentRoute.group.options.authentication;
+  if (authData.requireLoggedIn ^ authenticated()) {
+    FlowRouter.go( authData.redirectPath );
+    return true;
+  }
+  return false;
 };
 
 Template.default.helpers({
-	loggingIn() {
-		return Meteor.loggingIn();
-	},
-	authenticated() {
-		return !Meteor.loggingIn() && Meteor.user();
-	},
-	redirectAuthenticated() {
-	 	return handleRedirect([
-			'login',
-			'signup',
-			'recover-password',
-			'reset-password'
-		], '/' );
-	},
-	redirectPublic() {
-		return handleRedirect([
-			'index',
-			'dashboard'
-		], '/login' );
-	}
+  loggingIn() {
+    return Meteor.loggingIn();
+  },
+  authenticated() {
+    return authenticated();
+  },
+  redirectPublic() {
+    return handleRedirect(FlowRouter.current().route);
+  },
+  redirectAuthenticated() {
+    return handleRedirect(FlowRouter.current().route);
+  }
 });
```

Comment:
These two methods should be consolidated into one helper method because they essentially do the exact same thing. I have kept them the same to be as close to the original as possible while debugging my implementation.